File: C/Arreglos/Sort/ARR_ORD_2.c
//Selection Sort

#include <stdio.h>
#include <stdlib.h> 
#include <time.h> 

void Swap(int *a, int *b) 
{
   int temp = *a;
   *a = *b;
   *b = temp;
}

void CARGA_VECTOR(int *v) 
{
	int i;

	//printf("%p\n", v);

	for(i = 0; i < 10; ++i) 
	{		
		//Genero numeros entre 0 y 49
		v[i] = rand() % 50;
	}
}


void ORDENAR_VECTOR(int *v) 
{
	int j, k;

	for (j = 0; j < 9; j++)
	{

		int Min = j;
		
		//Busco un nuevo Min
		for (k = j + 1; k < 9; k++)
		{
			if (v[k] < v[Min])
			{
				Min = k;
			}
		}

		//Si encuentro un nuevo min lo cambio
		//if (Min != j) 
		//{
			Swap(&v[j], &v[Min]);
		//}
	}	
}


int main() 
{	
	int i;
	int V[10];

	// Uso el tiempo actual para el generador de numeros 
    srand(time(0)); 

	CARGA_VECTOR(V);

	printf(" ---------------------\n");
	printf(" ----- RESULTADOS ----\n");
	printf(" ---------------------\n");

	printf("Vector origen: \n");

	//valores de los elementos del vector
	for(i = 0; i < 10; ++i) {
		printf("%d ", V[i]);
	}

	printf("\n---------------------\n");

	ORDENAR_VECTOR(V);

	printf("Vector ordenado: \n");

	//valores de los elementos del vector
	for(i = 0; i < 10; ++i) {
		printf("%d ", V[i]);
	}

	return 0;
}

File: Python/Arreglos/Sort/ARR_ORD_2.py
import random

def Swap(a, b, i, j):
    a[i], b[j] = b[j], a[i]

def CARGA_VECTOR(v):
    for i in range(len(v)):
        v[i] = random.randint(0, 49)

def ORDENAR_VECTOR(v):
    n = len(v)
    for j in range(n-1):
        Min = j
        for k in range(j+1, n):
            if v[k] < v[Min]:
                Min = k
        # swap
        v[j], v[Min] = v[Min], v[j]

def main():
    V = [0]*10
    random.seed()
    CARGA_VECTOR(V)
    print(" ---------------------")
    print(" ----- RESULTADOS ----")
    print(" ---------------------")
    print("Vector origen: ")
    for val in V:
        print(val, end=' ')
    print()
    print("---------------------")
    ORDENAR_VECTOR(V)
    print("Vector ordenado: ")
    for val in V:
        print(val, end=' ')
    print()

if __name__ == "__main":
    main()
